### PR TITLE
Expand developer documentation.

### PR DESCRIPTION
**Pull request by dvogel**
* Document repo organization in docs/development.md.
* Improve focus of README.md on quick-start documentation.
* Split out README.md content into docs/development.md and docs/testing.md.

### CODE REVIEW

File: docs/testing.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+There's a variety of automated tests that will be run when you submit a pull request. In order to streamline
+the process of reviewing changes, it would be best for you to ensure each of these test suites passes locally
+before submitting a pull request.
+
+### Unit Tests
+
+General tests specific to a component or simple interactions between related components are handled by our
+unit test suite. This uses the [jest framekwork](https://jestjs.io/).
+```
+npm run-script test:unit:ci
```
Comment (edpaget) [inline docs/testing.md]:
```suggestion
npm run-script test:unit
```

CI suppresses stdout and writes to a xml file which is probably not the easiest to read. We probably even want to suggest `npm run-script test:unit -- --watch` which will run unit tests on file change.

Comment (dvogel) [reply docs/testing.md]: > LGTM. Maybe we should update the Node v6+ statement, to say we only support Node v10+? Just to standardize team on the version with the longest support window.

I'm OK with that.

I was curious why it said v6+. I suspected maybe that's what was used in the build process or in the CI jobs. The `Dockerfile` used for the visual regression tests is based on a `node:8` image, which I'm assuming is node version 8. I wasn't able to track down the version used in the build process. Where is that determined?

Comment (kalilsn) [reply docs/testing.md]: The build happens in the docker container, so it's node:8. I've been running on 10 so I'm sure we could just update that.

File: README.md [hunk 2]
```diff
@@ -11,54 +11,24 @@ First install the dependencies:
...
-If you need to update your snapshots (after seeing a failing diff), run with the `-u` flag:
-```
-docker run --rm -itv "$PWD:/application" -v "/application/node_modules" developer-portal npm run test:visual -- -u
-```
-
-It's important to run these tests inside the docker container, or else the tests will report false negatives due to font differences.
-
-## Deploying
+At this point you should have a browser open, with the developer portal loaded. If you make changes to the
+portal, your browser should auto-reload the page.
```
Comment (kalilsn) [inline README.md]:
```suggestion
code, your browser should auto-reload the page.
```

File: docs/testing.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
+# Testing Guide for the Developer Portal
+
+## Manually Testing Local Changes
+
+When you make changes locally, testing those changes will often require overriding an environment variable.
+This is done by modifying your `.env` file (no envionment-specific suffix). For example, assuming you
```
Comment (kalilsn) [inline docs/testing.md]:
```suggestion
This is done by modifying your `.env` file (no environment-specific suffix). For example, assuming you
```

File: docs/testing.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+follow-ed the setup instructions in the README.md file, the OpenAPI documentation displayed in the portal will
```
Comment (kalilsn) [inline docs/testing.md]:
```suggestion
followed the setup instructions in the README.md file, the OpenAPI documentation displayed in the portal will
```

File: docs/testing.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+We also use jest, combined with [Pupeteer](https://github.com/GoogleChrome/puppeteer), to run end-to-end tests
+for accessibility issues. **NOTE: THESE ARE CURRENTLY NOT WORKING ON MASTER.**
+
+
+### Visual Regression Testing
+
+This project uses `jest-image-snapshot` for visual regression testing. Screenshots of the app are checked into git, and regenerated whenever a change is made. If they don't match, Jenkins will report a test failure. To run these tests locally, you'll first need to build the docker image:
+```
+docker build -t developer_portal .
```
Comment (kalilsn) [inline docs/testing.md]:
```suggestion
docker build -t developer-portal .
```

(my mistake, derek caught it yesterday(

File: docs/testing.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+General tests specific to a component or simple interactions between related components are handled by our
+unit test suite. This uses the [jest framekwork](https://jestjs.io/).
+```
+npm run-script test:unit:ci
+```
+
+
+### End-to-End Tests
+
+We also use jest, combined with [Pupeteer](https://github.com/GoogleChrome/puppeteer), to run end-to-end tests
```
Comment (kalilsn) [inline docs/testing.md]:
```suggestion
We also use jest, combined with [Puppeteer](https://github.com/GoogleChrome/puppeteer), to run end-to-end tests
```

File: docs/testing.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+If you are working on OpenAPI documentation in vets-api, you will need to point the portal at your local
+instance of [vets-api](https://github.com/department-of-veterans-affairs/vets-api#base-setup). You will do
+this by overriding the `REACT_APP_VETSGOV_SWAGGER_API` environment variable:
+```
+REACT_APP_VETSGOV_SWAGGER_API=http://localhost:3000
+```
+
+## Automated Testing
+
+There's a variety of automated tests that will be run when you submit a pull request. In order to streamline
```
Comment (kalilsn) [inline docs/testing.md]:
```suggestion
There are a variety of automated tests that will be run when you submit a pull request. In order to streamline
```

File: README.md [hunk 8]
```diff
@@ -11,54 +11,24 @@ First install the dependencies:
 npm install
 ```
 
-Next, before running the app, create a `.env` file with the following contents:
+Next, create a `.env` file with the following contents (don't worry what these mean yet):
```
Comment (kalilsn) [inline README.md]:
```suggestion
Next, create a `.env` file with the following contents (don't worry about what these mean yet):
```

**Comment by dvogel** [comment]
> Contents look great overall! Should make it much easier for people on other teams to make changes.

Thanks for catching all those little mistakes! I swear I proof-read it :)

**Pull request merged**